### PR TITLE
Fix Ghoul2 API on server and UI for legacy mods

### PR DESCRIPTION
**Pull request by ensiform**
Also code cleanup for CGame Ghoul2 API in legacy processing.

### CODE REVIEW

**Comment by ensiform** [comment]
Needs testing.

**Comment by erfg12** [comment]
Compiled, ran with clanmod v1.3.0beta, spawned 10 reelo bots, nothing crashed.

**Comment by ensiform** [comment]
@erfg12 Added new commit that may help with that though I am not sure.

File: codemp/client/cl_uiapi.cpp [hunk 1]
```diff
@@ -290,6 +290,7 @@ static qboolean CL_G2API_GetBoltMatrix_NoRecNoRot( void *ghoul2, const int model
 		return qfalse;
 	}
 
+	re->G2API_BoltMatrixReconstruction( qfalse );
```
Comment (xycaleth) [inline codemp/client/cl_uiapi.cpp]: I don't think this should be here... I think it might only in cgame because of... https://github.com/JACoders/OpenJK/blob/master/codemp/cgame/cg_players.c#L9252-L9258

File: codemp/rd-dedicated/G2_API.cpp [hunk 2]
```diff
@@ -649,7 +642,7 @@ int G2API_InitGhoul2Model(CGhoul2Info_v **ghoul2Ptr, const char *fileName, int m
 
 qboolean G2API_SetLodBias(CGhoul2Info *ghlInfo, int lodBias)
 {
-	if (ghlInfo)
+	if (G2_SetupModelPointers(ghlInfo))
```
Comment (xycaleth) [inline codemp/rd-dedicated/G2_API.cpp]: Why are these changed?

Comment (ensiform) [reply codemp/rd-dedicated/G2_API.cpp]: To be in sync with the original code and rd-vanilla.

File: codemp/client/cl_uiapi.cpp (hunk 1 above)
Comment (ensiform) [reply codemp/client/cl_uiapi.cpp]: It is how the original SDK behaves I believe. I was just trying to make it all in sync with original behavior.

Comment (ensiform) [reply codemp/client/cl_uiapi.cpp]: Please check with that
